### PR TITLE
User story 31 - Merchant Item Show, Merchant See View Item Link

### PR DESCRIPTION
**Pull request by MackHalliday**
- Merchant can see view item link on their dashboard page (merchant_admin and merchant_employee) 

- Created new controller: Merchant Item Controller, nested under the merchant file 
- Update routes with merchant/items route, prefixed as merchant_user_items

- Added to seed files to connect merchant users with merchant shop

- Tested live
- Rspec OK 
- SimpleCov at 99.9% (Missing sad path testing from other user story)

### CODE REVIEW

File: app/views/merchant/items/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                <td><%= item.description %></td>
+                <td><%= item.price %></td>
+                <td><%= item.inventory %></td>
+                <td><%= item.active? %></td>
+              </tr>
+          </section>
+        <% end %>
+      </tbody>
+    </table>
+</div>
```
Comment (leiyakenney) [inline app/views/merchant/items/index.html.erb]: Nice use of the bootstrap table layout!

File: db/seeds.rb [hunk 2]
```diff
@@ -36,15 +36,17 @@
...
               state: "PA",
               zipcode: "18501",
               email: "dwightkschrute@email.com",
               password: "IdentityTheftIsNotAJoke",
-              role: 1)
+              role: 1,
+              merchant: bike_shop)
 admin_user = User.create!(name: "Leslie Knope",
               address: "14 Somewhere Ave",
               city: "Pawnee",
```
Comment (leiyakenney) [inline db/seeds.rb]: It looks like we both updated the seeds file, but in different ways. You added a merchant at the end; I created the user off the merchant. They do the same thing, but whoever merges our pr's will have to decide which way we want to move forward with for continuity :)

File: spec/features/users/merchant_users/item_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+RSpec.describe "Merchant Dashboard" do
+  before :each do
+    @merchant_1 = create(:merchant)
+
+    @item_1 = @merchant_1.items.create!(attributes_for(:item))
+    @item_2 = @merchant_1.items.create!(attributes_for(:item))
+
+    @merchant_admin = create(:user, role: 1, merchant: @merchant_1)
+    @merchant_employee = create(:user, role: 2, merchant: @merchant_1)
+  end
```
Comment (leiyakenney) [inline spec/features/users/merchant_users/item_index_spec.rb]: Nice use of FactoryBot!

File: app/controllers/merchant/items_controller.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class Merchant::ItemsController < Merchant::BaseController
+  def index
+    @items = current_user.merchant.items
```
Comment (nancylee713) [inline app/controllers/merchant/items_controller.rb]: Nice nested relationships!

File: app/views/merchant/items/index.html.erb (hunk 1 above)
Comment (nancylee713) [reply app/views/merchant/items/index.html.erb]: This looks great! Maybe in future user stories, `Item ID` might need to be clickable

File: spec/features/users/merchant_users/item_index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    within "#item-#{@item_2.id}" do
+      expect(page).to have_content(@item_2.name)
+      expect(page).to have_content(@item_2.description)
+      expect(page).to have_content(@item_2.price)
+      expect(page).to have_content(@item_2.inventory)
+      expect(page).to have_content(@item_2.active?)
+    end
+  end
+
+  it 'merchant admin sees link to view shop items' do
```
Comment (nancylee713) [inline spec/features/users/merchant_users/item_index_spec.rb]: test name should change to `merchant employee`

Comment (MackHalliday) [reply spec/features/users/merchant_users/item_index_spec.rb]: Updated! thanks!

**Pull request merged**